### PR TITLE
AWS system test example_dynamodb_to_s3: add logs when retry

### PR DESCRIPTION
**Pull request by vincbeck**
Add logs in the system test `example_dynamodb_to_s3` to log the different attempts and retries

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by ferruzzi — APPROVED** [review]
Strange that this is still flaky, the logs should help.

**Review by vandonr-amz — APPROVED** [review]

File: tests/system/providers/amazon/aws/example_dynamodb_to_s3.py [hunk 1]
```diff
@@ -51,6 +55,8 @@
 @tenacity.retry(
     stop=tenacity.stop_after_attempt(5),
     wait=tenacity.wait_exponential(min=5),
+    before=before_log(log, logging.INFO),
+    before_sleep=before_sleep_log(log, logging.WARNING),
```
Comment (o-nikolas) [inline tests/system/providers/amazon/aws/example_dynamodb_to_s3.py:59]: Why WARN for one and INFO for the other?

Comment (vincbeck) [reply tests/system/providers/amazon/aws/example_dynamodb_to_s3.py:59]: `before` is before an attempt so nothing anormal here. `before_sleep` happens between a fail and a sleep, hence the warn here :) [See doc](https://tenacity.readthedocs.io/en/latest/#before-and-after-retry-and-logging)

Comment (o-nikolas) [reply tests/system/providers/amazon/aws/example_dynamodb_to_s3.py:59]: I guess in my head I would have only warned if it times out, but I guess you could argue that should be ERROR at that point, so this looks good to me!

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**